docs/shared/guides/dynamic-mfe-angular.md

```diff
@@ -0,0 +1,167 @@
+# Advanced Angular Micro Frontends with Dynamic Module Federation
+
+Dynamic Module Federation is a technique that allows an application to determine the location of its remote applications at runtime. It helps to achieve the use case of **“Build once, deploy everywhere”**.
+
+“Build once, deploy everywhere” is the concept of being able to create a single build artefact of your application and deploy it to multiple environments such as staging and production.
+
+The difficulty in achieving this with a Micro Frontend Architecture using Static Module Federation is that our Remote applications will have a different location (or URL) in each environment. Previously, to account for this, we would have had to specify the deployed location of the Remote applications and rebuild the application for the target environment.
+
+This guide will walk through how the concept of “Build once, deploy everywhere” can be easily achieved in a Micro Frontend Architecture that uses Dynamic Module Federation.
+
+## Prerequisites
+
+This guide follows the previous Angular Micro Frontend guide. It will use the same workspace that was created in that guide.
+If you have not completed it, you can find that guide [here](/guides/setup-mfe-with-angular).
+
+## Aim
+
+The aim of this guide is two-fold. We want to be able to
+Transform an existing Static Module Federation setup to use Dynamic Federation.
+To generate a new Micro Frontend application that uses Dynamic Federation.
+
+## What we’ll build
+
+To achieve the aims, we will do the following:
+Change the Dashboard application to use Dynamic Federation
+Generate a new Employee Dashboard application that will use Dynamic Federation
+It should use the existing Login application
+It should use a new Todo application
+
+## Converting the Dashboard application
+
+The Dashboard application is a Host application that loads the Remote applications at runtime based on their deployed locations when the application was built.
+
+We want to change this so that the Dashboard application can make a network request at runtime to find out the deployed locations of the Remote applications.
+
+There are 3 steps involved with this:
+Make a network request to fetch the locations of the Remote applications _(the Remote Definitions)_.
+Set the Remote Definitions so that webpack is aware of how to find the Remote applications when requested.
+Change how we load the Remote applications so that webpack can fetch any required files correctly.
```

Comment:
```suggestion
There are 3 steps involved with this:

- Make a network request to fetch the locations of the Remote applications _(the Remote Definitions)_.
- Set the Remote Definitions so that Webpack is aware of how to find the Remote applications when requested.
- Change how we load the Remote applications so that Webpack can fetch any required files correctly.
```